Yoast/javascript
# Remove the help center remnants

## Summary

* [algolia-search-box] removed the entire package
* [components] removed the help center remnants
* [yoast-components] removed the help center remnants

This PR can be summarized in the following changelog entry:

* Removed the components related to the Help Center and the AlgoliaSearcher package

## Relevant technical choices:

* This PR seeks to remove all the composite components, wrappers, and single components used exclusively for the Help Center
  * removes the VideoTutorial component
  * removes the HelpCenterButton component
  * removes the HelpCenter component
* Entirely removes the AlgoliaSearcher package
* Keeps generic, reusable, components like the Tabs and the YouTubeVideo

Note: `packages/helpers/tests/__snapshots__/A11yNoticeTest.js.snap` got modified when updating the snapshots: looks like it's something related to `A11yNotice` which isn't touched in this PR, maybe it's because of some snapshot that missed to be updated.

## Test instructions
Not sure how this PR can be tested other than manually:
- the Help Center isn't in the standalone app any longer
- linking the monorepo to Yoast SEO and building the plugin will error, pending https://github.com/Yoast/wordpress-seo/issues/13859
- manually inspect the codebase to check there are no references to the Help Center and Algolia Search, check also configuration files like package.json, eslint, babel, travis


* This PR affects the following parts of the plugin, which may require extra testing:
  * Help Center in the settings pages
  * The Knowledge base search within the Help Center

## UI changes
Note: I guess the actual UI changes should be noted on https://github.com/Yoast/wordpress-seo/issues/13859

## Quality assurance


Fixes https://github.com/Yoast/javascript/issues/401
